src/CodeGen_LLVM.cpp

```diff
@@ -2998,16 +3004,24 @@ void CodeGen_LLVM::visit(const For *op) {
 }
 
 void CodeGen_LLVM::visit(const Store *op) {
-    Value *val = codegen(op->value);
+    // Even on 32-bit systems, Handles are treated as 64-bit in
+    // memory, so convert stores of handles to stores of uint64_ts.
+    if (op->value.type().is_handle()) {
```

Comment:
It should work with vectors of handles too. I check that case in the updated handle test.